src/script.js

```diff
@@ -2,39 +2,66 @@
 
 const SERVER_URL = '{{{SERVER_URL}}}';
 const params = new URLSearchParams(window.location.search);
-const id = params.get('id');
+const videoId = params.get('id');
+let psid;
 
 function handleEvent(data, eventType) {
   const xhr = new XMLHttpRequest();
   xhr.open('POST', SERVER_URL);
   xhr.setRequestHeader('Content-Type', 'application/json');
-  xhr.send(JSON.stringify({ data, eventType }));
+  xhr.send(JSON.stringify({ data, eventType, psid }));
 }
 
-document.addEventListener('DOMContentLoaded', () => {
+function handleError(err, title, message) {
+  const div = document.createElement('div');
+  div.classList.add("error-container");
+  div.innerHTML = `<h1>${title}</h1><p>${message}</p>`;
+  document.querySelector('.container').innerHTML = ``
+  document.querySelector('.container').appendChild(div);
+  console.error(err);
+}
+
+function setPlayer() {
   const options = {
-    id,
-    width: 800
+    id: videoId,
+    responsive: true
   };
+
   const player = new Vimeo.Player('vimeoVideo', options);
 
   player.ready().then(() => {
     player.on('ended', data => handleEvent(data, 'ended'));
-  
+
     player.on('error', data => handleEvent(data, 'error'));
-    
+
     player.on('pause', data => handleEvent(data, 'pause'));
-    
+
     player.on('play', data => handleEvent(data, 'play'));
-    
+
     player.on('playbackratechange', data => handleEvent(data, 'playbackratechange'));
-    
+
     player.on('seeked', data => handleEvent(data, 'seeked'));
-  
+
     player.on('volumechange', data => handleEvent(data, 'volumechange'));
   }).catch((err) => {
-    console.error(err);
-    document.querySelector('body').innerHTML = `<p>Video not found</p>`;
+    const title = '❌ Not found';
+    const message = 'Sorry, we couldn’t find that video'
+    handleError(err, title, message);    
   });
+}
 
-});
+document.addEventListener('DOMContentLoaded', () => {
+  window.extAsyncInit = function () {
+    MessengerExtensions.getContext('{{{APP_ID}}}',
+      function success(thread_context) {
+        psid = thread_context.psid;
```

Comment:
Handle race conditions! 

What if the vimeo loads and starts sending events before the messenger extensions getContext returns? psid will not be known! This should resolve a promise and the vimeo player should wait for that promise before loading and sending events...

Reply:
But I'm not setting the player until I receive the `thread_context`.

Reply:
Hehehe, good point! :D